src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -1025,6 +1074,55 @@ private void ValidateCallerMemberNameAttribute(AttributeSyntax node, BindingDiag
+
+            if (!IsValidCallerInfoContext(node))
+            {
+                // CS8912: The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect because it applies to a
+                //         member that is used in contexts that do not allow optional arguments
+                diagnostics.Add(ErrorCode.WRN_CallerArgumentExpressionParamForUnconsumedLocation, node.Name.Location, CSharpSyntaxNode.Identifier.ValueText);
+            }
+            else if (!compilation.Conversions.HasCallerInfoStringConversion(TypeWithAnnotations.Type, ref useSiteInfo))
+            {
+                // CS8913: CallerArgumentExpressionAttribute cannot be applied because there are no standard conversions from type '{0}' to type '{1}'
+                TypeSymbol stringType = compilation.GetSpecialType(SpecialType.System_String);
+                diagnostics.Add(ErrorCode.ERR_NoConversionForCallerArgumentExpressionParam, node.Name.Location, stringType, TypeWithAnnotations.Type);
+            }
+            else if (!HasExplicitDefaultValue && !ContainingSymbol.IsPartialImplementation()) // attribute applied to parameter without default
+            {
+                // Unconsumed location checks happen first, so we require a default value.
+
+                // CS8914: The CallerArgumentExpressionAttribute may only be applied to parameters with default values
+                diagnostics.Add(ErrorCode.ERR_BadCallerArgumentExpressionParamWithoutDefaultValue, node.Name.Location);
+            }
+            else if (HasCallerLineNumberAttribute)
+            {
+                // CS8915: The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect. It is overridden by the CallerLineNumberAttribute.
+                diagnostics.Add(ErrorCode.WRN_CallerLineNumberPreferredOverCallerArgumentExpression, node.Name.Location, CSharpSyntaxNode.Identifier.ValueText);
+            }
+            else if (HasCallerFilePathAttribute)
+            {
+                // CS8916: The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect. It is overridden by the CallerFilePathAttribute.
+                diagnostics.Add(ErrorCode.WRN_CallerFilePathPreferredOverCallerArgumentExpression, node.Name.Location, CSharpSyntaxNode.Identifier.ValueText);
+            }
+            else if (HasCallerMemberNameAttribute)
+            {
+                // CS8917: The CallerArgumentExpressionAttribute applied to parameter '{0}' will have no effect. It is overriden by the CallerMemberNameAttribute.
+                diagnostics.Add(ErrorCode.WRN_CallerMemberNamePreferredOverCallerArgumentExpression, node.Name.Location, CSharpSyntaxNode.Identifier.ValueText);
+            }
+            else if (attribute.CommonConstructorArguments.Length == 1 &&
+                GetEarlyDecodedWellKnownAttributeData()?.CallerArgumentExpressionParameterIndex == -1)
```

Comment:
>?. [](start = 55, length = 2)

Is the conditional access necessary?

Reply:
> > ?. [](start = 55, length = 2)
> 
> Is the conditional access necessary?

I'm not really sure. Most existing callers do that. 😕

Reply:
>> Is the conditional access necessary?

> I'm not really sure. Most existing callers do that. 😕 

Not a big deal, it is cheap to keep it. It looks like ```EarlyDecodeWellKnownAttribute``` is going to allocate the object though.

---
In reply to: [601878392](https://github.com/dotnet/roslyn/pull/51952#discussion_r601878392) [](ancestors = 601878392)